lib/ts/recipe/webauthn/types.ts

```diff
@@ -0,0 +1,190 @@
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import type {
+    FeatureBaseConfig,
+    NormalisedBaseConfig,
+    NormalisedFormField,
+    NormalisedGetRedirectionURLContext,
+    UserContext,
+    WebJSRecipeInterface,
+} from "../../types";
+import type {
+    OnHandleEventContext as AuthRecipeModuleOnHandleEventContext,
+    UserInput as AuthRecipeModuleUserInput,
+    NormalisedConfig as NormalisedAuthRecipeModuleConfig,
+    Config as AuthRecipeModuleConfig,
+} from "../authRecipe/types";
+import type { InputProps } from "../emailpassword/components/library/input";
+import type WebJSRecipe from "supertokens-web-js/recipe/webauthn";
+import type { RecipeInterface } from "supertokens-web-js/recipe/webauthn";
+import type { User } from "supertokens-web-js/types";
+
+export type WebauthnFeatureBaseConfig = {
+    disableDefaultUI?: boolean;
+} & FeatureBaseConfig;
+
+export type GetRedirectionURLContext = NormalisedGetRedirectionURLContext<{
+    /*
+     * Get Redirection URL Context
+     */
+    action: "RECOVER_ACCOUNT";
+}>;
+
+export type PreAndPostAPIHookAction =
+    | "REGISTER_OPTIONS"
+    | "SIGN_IN_OPTIONS"
+    | "SIGN_UP"
+    | "SIGN_IN"
+    | "EMAIL_EXISTS"
+    | "GENERATE_RECOVER_ACCOUNT_TOKEN"
+    | "RECOVER_ACCOUNT";
+
+export type OnHandleEventContext =
+    | {
+          action: "SUCCESS";
+          isNewRecipeUser: boolean;
+          createdNewSession: boolean;
+          user: User;
+      }
+    | {
+          action: "REGISTER_OPTIONS";
+      }
+    | {
+          action: "SIGN_IN_OPTIONS";
+      }
+    | {
+          action: "GET_EMAIL_EXISTS";
+          exists: boolean;
+      }
+    | {
+          action: "REGISTER_CREDENTIAL";
+      }
+    | {
+          action: "AUTHENTICATE_CREDENTIAL";
+      }
+    | {
+          action: "SIGN_IN";
+      }
+    | {
+          action: "SIGN_UP";
+      }
+    | {
+          action: "GENERATE_RECOVER_ACCOUNT_TOKEN";
+      }
+    | {
+          action: "RECOVER_ACCOUNT";
+      }
+    | {
+          action: "REGISTER_CREDENTIAL_WITH_SIGN_UP";
+      }
+    | {
+          action: "AUTHENTICATE_CREDENTIAL_WITH_SIGN_IN";
+      }
+    | {
+          action: "REGISTER_CREDENTIAL_WITH_RECOVER_ACCOUNT";
+      }
+    | {
+          action: "FAILED_TO_REGISTER_USER";
+      }
+    | AuthRecipeModuleOnHandleEventContext;
+
+export type UserInput = Record<string, unknown> & {
+    override?: {
+        functions?: (originalImplementation: RecipeInterface) => RecipeInterface;
+    };
+    signUpFeature?: NormalisedSignUpFormFeatureConfig;
```

Comment:
We need a similar input prop for the account recovery feature.
Do we need a similar one for sign in as well or shouldn't this be `signInAndUpFeature`?